src/oauth2.js

```diff
@@ -35,33 +35,57 @@ function login (req, res) {
   });
 }
 
+function isDashAdmin (user) {
+  try {
+    const guild = bot.guilds.get(config.mainGuildId);
+    const member = guild.members.get(user.id);
+    return utils.isAdmin(member);
+  } catch (err) {
+    return false;
+  }
+}
+
+async function getAuthUser (token) {
+  let user;
+
+  try {
+    let accessToken = jwt.verify(token, config.clientSecret).token;
+    let response = await superagent.get("https://discord.com/api/users/@me")
+      .set("Authorization", `Bearer ${accessToken}`);
+
+    user = response.body;
+  } catch (err) {
+    return;
+  }
+
+  return user;
+}
+
 async function checkAuth (req, res, next) {
   if (req.cookies.token) {
-    let token = req.cookies.token;
-    try {
-      let accessToken = jwt.verify(token, config.clientSecret).token;
-      let response = await superagent.get("https://discord.com/api/users/@me")
-      .set("Authorization", `Bearer ${accessToken}`);
-      req.user = response.body;
-    } catch (err) {
-      return res.redirect("/login");
-    }
+    req.user = await getAuthUser(res.cookies.token);
```

Comment:
should this be req instead of res, for the cookie?

Reply:
Whoops lol, tysm